### PR TITLE
Major Additions to support proper build, test, and CI

### PR DESCRIPTION
**Pull request by unlikelyzero**
Closes: #128 #191 #193 #194 #195 #14

### Describe your changes:
- [ ] webpack-dev-server proxy configuration is incompatible with quickstart in CI
- [x] not necessary: fix inability to archive the makeout.txt at end of run. Some sort of pathing issue
- [x] Remove package-lock.json file to match openmct
- [x] Set pinned dependencies to let existing dependabot implementation manage revisions
- [x] Updates dependencies to match openmct
- [x] Add basic e2e suite execution with github actions
- [x] Add minimum supported node version to package.json
- [x] Add `npm run build:example` to general CI checks
- [x] Add minimum supported openmct version to our package.json

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Unit tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [ ] Testing instructions included in associated issue?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate unit tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Commit messages meet standards?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: .github/workflows/yamcs-quickstart-e2e.yml [hunk 1]
```diff
@@ -7,30 +7,48 @@ on:
...
-      - run: ls -latr
+      - run: npx playwright@1.25.2 install chromium
       - run: npm run build:example
       - run: npm run test:getopensource
       - run: npm run test:e2e:quickstart
+      - name: Archive makeout.txt #not working due to pathing
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          path: /quickstart/docker/makeout.txt
```
Comment (ozyx) [inline .github/workflows/yamcs-quickstart-e2e.yml:44]: maybe needs a relative path here?

File: tests/e2e/playwright-quickstart.config.js [hunk 2]
```diff
@@ -14,7 +14,8 @@ const config = {
         trace: 'on',
         baseURL: 'http://localhost:9000/#',
         ignoreHTTPSErrors: true,
-        myItemsFolderName: "My Items"
+        myItemsFolderName: "My Items",
+        failOnConsoleError: false
```
Comment (ozyx) [inline tests/e2e/playwright-quickstart.config.js:18]: do we want to keep this disabled for yamcs quickstart testing? or leave it on and disable on a suite-by-suite basis as we have done with the couchDB tests?

Comment (unlikelyzero) [reply tests/e2e/playwright-quickstart.config.js:18]: Let's do the latter

File: tests/README.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# e2e Testing
+This project is using Open MCT's e2e-as-a-dependency model. To learn more, please see the official documentation on the [Official README](https://github.com/nasa/openmct/blob/master/e2e/README.md)
+
+# How to Run locally
+1. git clone yamcs/quickstart
```
Comment (scottbell) [inline tests/README.md:5]: we should add some backticks, e.g., `git clone yamcs/quickstart`

File: webpack.config.js [hunk 4]
```diff
@@ -68,13 +68,13 @@ const WEBPACK_CONFIG = {
         }],
         proxy: {
             "/yamcs-proxy/*": {
-                target: "http://localhost:8090/",
+                target: "http://127.0.0.1:8090/",
```
Comment (scottbell) [inline webpack.config.js:71]: why change this from `localhost`?

**Review by scottbell — COMMENTED** [review]
Looking good @unlikelyzero! I had a few small comments

File: .github/workflows/yamcs-quickstart-e2e.yml [hunk 5]
```diff
@@ -7,30 +7,48 @@ on:
...
         with:
           repository: yamcs/quickstart
           ref: master
-      - run: make all
+          path: 'quickstart'
+      - name: trigger make all from docker dir. Orphan and redirect output to docker/makeout.txt
+        run: |
+          cd quickstart/docker
+          touch makeout.txt
+          make all &> makeout.txt &
```
Comment (ozyx) [inline .github/workflows/yamcs-quickstart-e2e.yml:23]: 💋

File: .github/workflows/yamcs-quickstart-e2e.yml (hunk 1 above)
Comment (ozyx) [inline .github/workflows/yamcs-quickstart-e2e.yml:44]: Remove the prepended `/` here

File: tests/e2e/playwright-quickstart.config.js (hunk 2 above)
Comment (ozyx) [inline tests/e2e/playwright-quickstart.config.js:18]: Let's remove this so it's enabled by default and we can disable on a suite-by-suite basis

**Review by ozyx — CHANGES_REQUESTED** [review]
Suggesting a couple changes but otherwise looks good to me

File: webpack.config.js [hunk 6]
```diff
@@ -68,13 +68,13 @@ const WEBPACK_CONFIG = {
...
                 secure: false,
                 changeOrigin: true,
                 pathRewrite: {'^/yamcs-proxy/': ''}
             },
             "/yamcs-proxy-ws/*": {
-                target: "ws://localhost:8090/api/websocket",
+                target: "ws://127.0.0.1:8090/api/websocket",
```
Comment (scottbell) [inline webpack.config.js:77]: Same question here

Comment (unlikelyzero) [reply webpack.config.js:77]: @scottbell our webpack-dev-server proxy is not working in CI https://github.com/akhenry/openmct-yamcs/issues/191

File: .github/workflows/yamcs-quickstart-e2e.yml (hunk 1 above)
Comment (unlikelyzero) [reply .github/workflows/yamcs-quickstart-e2e.yml:44]: even that doesn't help. There's something being destroyed when cloning the second time

File: webpack.config.js (hunk 6 above)
Comment (scottbell) [reply webpack.config.js]: We think it's due to us not waiting for YAMCS to come up before starting tests

**Review by scottbell — APPROVED** [review]

**Review by ozyx — APPROVED** [review]
LGTM

**Pull request merged**